Backport the upstream fix limiting the chunk-extension search to the bytes
actually received.

Upstream commit:
https://gitlab.gnome.org/GNOME/libsoup/-/commit/72e566d82363ba35cdcf4a8c7b4826029f7190ff

--- a/libsoup/http1/soup-body-input-stream.c
+++ b/libsoup/http1/soup-body-input-stream.c
@@ -198,7 +198,7 @@
 		}
 
                 /* ignore extensions */
-                boundary = strstr (metabuf, ";");
+                boundary = memchr (metabuf, ';', nread);
                 if (boundary)
                         *boundary = '\0';
                 else
